### PR TITLE
Create model with associations and features

### PR DESCRIPTION
**Pull request by ZeenatLawal**
In this milestone we added:
The user is able to send a friendship invitation.
The user is able to see a button “Invite to friendship” next to the name of the user who is not my friend yet - on both users’ list and the single-user page.
The user is able to see pending friendship invitations sent to me from other users.
The user is able  to accept or reject friendships invitation.

### CODE REVIEW

**Comment by igorkol91** [comment]
The previous milestone was approved as shown in this link we are providing https://github.com/igorkol91/ror-social-scaffold/pull/2.
Unfortunately, we still have 'changes requested' in the dashboard and although we emailed the support we still didn't get any answer.

File: app/views/users/_user.html.erb
Comment (SajjadAhmad14) [inline app/views/users/_user.html.erb]: Remove this logic and add to app/helpers. [OPTIONAL]

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] You have mastered Git Flow.
- [x] PR title and summary is descriptive

## To Improve 🙏
- [x] User1 sends friendship request to user2, but user2 is not able to see incoming request. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png).
- [x] User should be able to accept friend request. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png).
- [x] User should be able to reject friend request. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png). 
- [x] Posts timeline should include posts written by logged-in user and their friends. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png).
- [x] Please add integration tests for friendship functionality.
- [x] Please see inline comments for reference.
# Optional
- [ ] Views should have no logic. You can use app/helpers to put all of your view-related logic and render it in views.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

File: app/views/users/show.html.erb
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb]: Remove this logic and add to app/helpers. [OPTIONAL]

**Comment by ZeenatLawal** [comment]
A logged-in user can see their received and pending friend requests only for their accounts on their profile page. We assumed the previous reviewer missed this point that's why we are explaining this.:smile:

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coders, 👋🏽 
**This is a great project you have going on, well-done lads and thanks for implementing the changes requested earlier on**. 

### Well-done 💯 
    1. Users can now accept and reject requests.

**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

### REQUIRED

- [x] Posts timeline should include posts written by logged-in user and their friends. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png). Right now, a user has all posts on their index page from all users even when they are not friends.

### OPTIONAL

- [x] Once a friend request has been sent, accepted or rejected, the `Add friend` button still exists on the all users page. This isn't the expected app's behavior.
- [ ] Please add integration tests for the app's functionality. Place these tests in the features directory. 

#

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

<hr>

Mention @mke2111 in this thread and I'll be glad to help. 😄 
### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Comment by mke2111** [comment]
> The previous milestone was approved as shown in this link we are providing #2.
> Unfortunately, we still have 'changes requested' in the dashboard and although we emailed the support we still didn't get any answer.

Has this issue been solved? If not, reach out to the reviewer that approved that milestone. They'll know what to do and your milestone will be marked complete within a few hours.

Best regards.

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coders, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The Posts timeline now has the expected behavior.

### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

<hr>

### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Pull request merged**
